agilent_infiniium_scope.py

```diff
@@ -409,67 +438,69 @@ def _parsePreamble(preamble):
-                ('yUnits',True),
-                ('maxBW',True),
-                ('minBW',True)]
-    preambleDict = {}
-    for key,val in zip(preambleKeys,preambleVals):
+    preamble_keys = [('byteFormat', True),
+                ('dataType', True),
+                ('numPoints', True),
+                ('count', True),
+                ('xStep', True),
+                ('xFirst', True),
+                ('xRef', True),
+                ('yStep', True),
+                ('yOrigin', True),
+                ('yRef', True),
+                ('coupling', True),
+                ('xRange', True),
+                ('xLeftDisplay', True),
+                ('yRange', True),
+                ('yCenterDisplay', True),
+                ('date', True),
+                ('time', True),
+                ('model', True),
+                ('acquisitionMode', True),
+                ('percentTimeBucketsComplete', True),
+                ('xUnits', True),
+                ('yUnits', True),
+                ('maxBW', True),
+                ('minBW', True)]
+    preample_dict = {}
+    for key, val in zip(preamble_keys, preamble_vals):
         if key[1]:
-            preambleDict[key[0]] = val
-    def unitType(num):
-        if num=='1':
+            preample_dict[key[0]] = val
+
+    def unit_type(num):
+        if num == '1':
             return 'V'
-        elif num=='2':
+        elif num == '2':
             return 'ns'
         else:
             raise Exception('Units not time or voltage')
-    preambleDict['xUnit'] = unitType(preambleDict['xUnits'])
-    preambleDict['yUnit'] = unitType(preambleDict['yUnits'])
-    return (preambleDict)
+    preample_dict['xUnit'] = unit_type(preample_dict['xUnits'])
+    preample_dict['yUnit'] = unit_type(preample_dict['yUnits'])
+    return (preample_dict)
+
 
-def _parseBinaryData(data, wordLength):
+def _parseBinaryData(data, word_length):
     """Parse binary data packed as string of RIBinary
     """
-    formatChars = {'1':'b','2':'h', '4':'f'}
-    formatChar = formatChars[str(wordLength)]
-
-    #Get rid of header crap
-    #unpack binary data
-    if wordLength == 1:
-        lenHeader = int(data[1])
-        dat = data[(2+lenHeader):]
-        dat = numpy.array(unpack(formatChar*(len(dat)/wordLength),dat))
-    elif wordLength == 2:
-        lenHeader = int(data[1])
-        dat = data[(2+lenHeader):]
-        dat = dat[-calcsize('>' + formatChar*(len(dat)/wordLength)):]
-        dat = numpy.array(unpack('>' + formatChar*(len(dat)/wordLength),dat))
-    elif wordLength == 4:
-        lenHeader = int(data[1])
-        dat = data[(2+lenHeader):]
-        dat = dat[-calcsize('>' + formatChar*(len(dat)/wordLength)):]
-        dat = numpy.array(unpack('>' + formatChar*(len(dat)/wordLength),dat))      
+    format_chars = {'1': 'b', '2': 'h', '4': 'f'}
+    format_char = format_chars[str(word_length)]
+
+    # Get rid of header
+    # unpack binary data
+    if word_length == 1:
```

Comment:
I might recommend not printing the actual data in the exception.  The expected byte string is many tens of thousands of bytes, so you'll surely miss the important part of the message if you print it all.

Is "//" a real thing or a typo for "/" here:

`num_words = len_wave // word_length`